Google-Summer-of-Code-2023.md

```diff
@@ -273,3 +273,504 @@ The following is a list of Oppia's 2023 GSoC project ideas. You are welcome to c
+  - For long-running tests:
+    - If the test is essential, find a way to make it shorter.
+    - If it is not, remove it from CI and put it in an acceptance layer that only runs on some release-candidate builds (see below). 
+- Split off less-important long-running tests into acceptance tests:
+- Create a GitHub Action that, at regular time intervals, takes the latest release-candidate build on develop that passes all CI checks, runs the acceptance tests on them. Builds that pass all acceptance tests should have a public, machine-readable indication that they did, and builds that cause an acceptance test failure should result in an email or chat notification to the CI Stability team.
+
+**Size of this project:** large (~350 hours)
+
+**Potential Mentors:** 
+
+**Knowledge/Skills Recommended:**
+
+**Suggested Milestones:**
+
+* **Milestone 1:** CI checks run in under an hour. Less important tests are split off into an acceptance layer.
+
+* **Milestone 2:** Pre-push checks run in under 5 minutes.
+
+**Dependency on Release Schedule:** 
+
+**Proposal notes:**
+
+**Useful resources:**
+
+### 4.5. Normalize Usage of Feature Flags
+
+**Project Description:**
+
+The aim of this project is to normalize the usage of feature flags (implemented using [platform parameters](https://github.com/oppia/oppia/wiki/Developing-new-features-with-feature-gating)) in Oppia’s deployment process. At the end of the project:
+
+- The feature flags dashboard should be moved from /admin to /release-coordinator, and be made more intuitive. (The current UI is not intuitive at all – it’s quite unclear what to do.) The common case of switching a boolean flag on and off should be easy for developers to do.
+- Feature flags should support rollbacks. There seems to be an [issue](https://github.com/oppia/oppia/issues/16416) with rollbacks not being correctly captured that should be diagnosed and fixed.
+- Extend the current functionality to allowlist specific usernames for external trusted tester programs.
+- Ensure the documentation is sufficiently complete so that developers are used to creating feature flags for new features, and have no questions about how to do so.
+
+**Size of this project:** medium (~175 hours)
+
+**Potential Mentors:** 
+
+**Knowledge/Skills Recommended:** Python and TypeScript
+
+**Suggested Milestones:**
+
+* **Milestone 1:** The feature flags dashboard is moved from /admin to /release-coordinator. It is also made more intuitive; in particular, the common case of switching a boolean flag on and off is easy for developers to do.  Feature flags support rollbacks (see [#16416](https://github.com/oppia/oppia/issues/16416)).
+
+* **Milestone 2:** Feature flags support allowlisting specific user IDs for external trusted tester programs. The documentation for handling feature flags is fully complete and developers use feature flags as a matter-of-course, without any issues.
+
+**Dependency on Release Schedule:** 
```

Comment:
There is no dependency, unless the applicant proposes using beam jobs (which I don't think they'll need)